src/datasets/summarizations/trend-weekly-elaboration.summarization.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { queryFactory } from './trend-weekly-elaboration.summarization';
+import { TimeSeriesPoint } from '../metas/types';
+import { hasHighValidity, hasLowValidity, isTextPartsInSummary, norSummaryFilters } from './utils/tests';
+
+describe('queryFactory', () => {
+
+  it('should have validity value between 0 and 1.', () => {
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 1; i <= 31; i++) {
+      const x = new Date(2020, 6, i);
+      const y = i;
+      points.push({ x, y });
+    }
+
+    const summaries = queryFactory(points)();
+    for (const summary of summaries) {
+      expect(summary.validity).toBeGreaterThanOrEqual(0.0);
+      expect(summary.validity).toBeLessThanOrEqual(1.0);
+    }
+  });
+
+  it('should create summaries describing monotonic linearly increasing data.', () => {
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 6; i <= 26; i++) {
+      const x = new Date(2020, 6, i);
+      const y = i * 10;
+      points.push({ x, y });
+    }
+
+    const validSummaryFilters = [
+      isTextPartsInSummary('first week', 'Monday to Sunday', 'increased'),
+      isTextPartsInSummary('second week', 'Monday to Sunday', 'increased'),
+      isTextPartsInSummary('third week', 'Monday to Sunday', 'increased'),
+    ];
+    const isOtherSummary = norSummaryFilters(...validSummaryFilters);
+
+    const summaries = queryFactory(points)();
+    const validSummaries = validSummaryFilters.map(summaryFilter =>
+      summaries.filter(summaryFilter)
+    );
+    const otherSummaries = summaries.filter(isOtherSummary);
+
+    for (const validSummary of validSummaries) {
+      expect(validSummary.length).toBe(1);
+      expect(validSummary.every(hasHighValidity)).toBeTrue();
+    }
+    expect(otherSummaries.every(hasLowValidity)).toBeTrue();
+  });
+
+  it('should create summaries describing monotonic linearly decreasing data.', () => {
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 6; i <= 26; i++) {
+      const x = new Date(2020, 6, i);
+      const y = 1000 - i * 10;
+      points.push({ x, y });
+    }
+
+    const validSummaryFilters = [
+      isTextPartsInSummary('first week', 'Monday to Sunday', 'decreased'),
+      isTextPartsInSummary('second week', 'Monday to Sunday', 'decreased'),
+      isTextPartsInSummary('third week', 'Monday to Sunday', 'decreased'),
+    ];
+    const isOtherSummary = norSummaryFilters(...validSummaryFilters);
+
+    const summaries = queryFactory(points)();
+    const validSummaries = validSummaryFilters.map(summaryFilter =>
+      summaries.filter(summaryFilter)
+    );
+    const otherSummaries = summaries.filter(isOtherSummary);
+
+    for (const validSummary of validSummaries) {
+      expect(validSummary.length).toBe(1);
+      expect(validSummary.every(hasHighValidity)).toBeTrue();
+    }
+    expect(otherSummaries.every(hasLowValidity)).toBeTrue();
+  });
+
+  it('should create summaries describing weekdays high increasing weekends low data.', () => {

```

Comment:
what does "low data" here and "high data" in the test below mean?

Reply:
"weekdays high weekends low" means weekdays and weekends have averages not close, and weekdays are high and weekends are low.

Reply:
'should create summaries describing data where weekdays are high and weekends are low'